apis/s3/src/test/java/org/jclouds/s3/S3ClientLiveTest.java

```diff
@@ -688,4 +691,26 @@ public void testUpdateObjectCannedACL() throws Exception {
          returnContainer(containerName);
       }
    }
+
+   public void testList_EmptyOptionSingleContainer() throws Exception {
+      String containerName = getContainerName();
+      try {
+         S3Object object = getApi().newS3Object();
+         object.getMetadata().setKey("a");
+         object.setPayload(TEST_STRING);
+         getApi().putObject(containerName, object);
+         // Test listing where we set the prefix and delimiter to empty string
+         ListBucketResponse rs = getApi().listBucket(containerName,
+               ListBucketOptions.Builder.delimiter("").withPrefix("").afterMarker(""));
+         assertEquals(rs.size(), 3);
+         Set<String> expected = Sets.newHashSet("a");
```

Comment:
Yes -- will combine these. I wasn't sure about other providers' behavior, but I suspect it should either match or ignore the test.